packages/services/src/connectionstatus.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+
+import { IConnectionStatus } from './tokens';
+
+/**
+ * Application connection status
+ */
+export class ConnectionStatus implements IConnectionStatus {
```

Comment:
In the previous version of this PR, `ConnectionStatus` and `IConnectionStatus` were part of the `@jupyterlab/application` package, and its related plugin part of `@jupyterlab/application-extension`.

With the current state of this PR, it is part of `@jupyterlab/services`. Although this may be more related to an "application", having it in `@jupyterlab/services` and its related plugin in `@jupyterlab/services-extension` has two advantages:

- no dependency on `@jupyterlab/application` from `@jupyterlab/services-extension`
- all `ServiceManagerPlugin` plugin definitions are in `@jupyterlab/services-extension`

Reply:
What if `@jupyterlab/application` imports it from services and re exports it and deprecates it (optionally)? Since it's a static const it shouldn't matter where you get it from I think?

Reply:
I meant in the previous attempt, for example in https://github.com/jupyterlab/jupyterlab/pull/16556, which was closed by not merged.

Currently in JupyterLab there is no `ConnectionStatus` / `IConnectionStatus`, so not sure we need to re-export? So it was mostly a comment on where the new `ConnectionStatus` class and `IConnectionStatus` interface and token should live.